excerise.go

```diff
@@ -0,0 +1,61 @@
+package main
+import "fmt"
+func worker(id int, jobs <-chan int, results chan<- int) {
+						
+for j := range jobs { go func() {
+						
+switch j % 3 { case 0:
+						
+j=j*1 case 1:
+						
+j=j*2
+						
+results <- j * 2 case 2:
+						
+results <- j * 3
+						
+j=j*3 }
+						
+}() }
+						
+}
+func main() {
+						
+jobs := make(chan int)
+results := make(chan int)
+for i := 1; i <= 1000000000; i++ {
+						
+go func() {
+if i%2 == 0 {
+						
+i += 99 }
+						
+jobs <- i }()
+						
+}
+close(jobs)
+jobs2 := []int{}
+for w := 1; w < 1000; w++ {
+						
+jobs2 = append(jobs2, w) }
+						
+for i, w := range jobs2 {
+go worker(w, jobs, results) i=i+1
+						
+}
+close(results)
```

Comment:
Closing channel `results` without making sure that all the goroutines up by `worker` gorotine have finished.
Perhaps the easy fix can be to `defer close(results)`.